reference-implementation/lib/writable-stream.js

```diff
@@ -332,15 +315,16 @@ function WritableStreamHandleAbortRequestIfPending(stream) {
     return;
   }
 
-  WritableStreamFinishAbort(stream);
-
   const abortRequest = stream._pendingAbortRequest;
   stream._pendingAbortRequest = undefined;
   const promise = WritableStreamDefaultControllerAbort(stream._writableStreamController, abortRequest._reason);
   promise.then(
     abortRequest._resolve,
     abortRequest._reject
   );
+  stream._backpressure = false;
```

Comment:
I think I have a fix that works by moving a tiny bit of the logic back out of WritableStreamError() and back to all the call sites except for WritableStreamFinishAbort. Uploading now-ish.

Upon further consideration it's not so bad to use _pendingAbortRequest as such a signal. It's basically a stand-in for a state between "writable" and "errored" ("aborting" perhaps). And such a state is a reasonable thing to consult when talking about messing with the ready promise.